android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/home/HomeScreen.kt

```diff
@@ -1,23 +1,52 @@
 package vn.luongvo.kmm.survey.android.ui.screens.home
 
 import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
 import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.ExperimentalLifecycleComposeApi
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import com.google.accompanist.pager.*
 import org.koin.androidx.compose.getViewModel
 import vn.luongvo.kmm.survey.android.ui.common.DimmedImageBackground
 import vn.luongvo.kmm.survey.android.ui.screens.home.views.*
 import vn.luongvo.kmm.survey.android.ui.theme.AppTheme
 import vn.luongvo.kmm.survey.android.ui.theme.ComposeTheme
+import vn.luongvo.kmm.survey.android.util.userReadableMessage
 
+const val HomeUserAvatar = "HomeUserAvatar"
+
+@OptIn(ExperimentalLifecycleComposeApi::class)
 @Composable
 fun HomeScreen(
     viewModel: HomeViewModel = getViewModel()
 ) {
+    val currentDate by viewModel.currentDate.collectAsStateWithLifecycle()
+    val avatarUrl by viewModel.avatarUrl.collectAsStateWithLifecycle()
+
+    val scaffoldState: ScaffoldState = rememberScaffoldState()
+    val context = LocalContext.current
+    LaunchedEffect(viewModel.error) {
+        viewModel.error?.let {
+            scaffoldState.snackbarHostState.showSnackbar(
+                message = it.userReadableMessage(context)
+            )
+            viewModel.error = null
```

Comment:
we should not modify the viewmodel's variable directly in Compose 🤔 I understand why you do that with this approach, but we should avoid using it IMO. How about [this approach](https://stackoverflow.com/a/69354712)? 🙆‍♂️ Feel free to discuss to find a good solution for this 🙇

Reply:
@manh-t I replied you on Slack instead https://nimble-co.slack.com/archives/C04BVRP6UDR/p1672386758164719 👋

Reply:
@manh-t Fixed in https://github.com/luongvo/kmm-survey/pull/62/commits/74ac75a08423c5ff5a73d5fc68402b6e316de020